src/BugsnagUnity/IDelivery.cs

```diff
@@ -116,25 +116,41 @@ IEnumerator PushToServer(IPayload payload, byte[] body)
                 {
                     yield return new WaitForEndOfFrame();
                 }
-                if (req.responseCode >= 200 && req.responseCode < 300)
+
+                if (IsSuccess(req.responseCode))
                 {
                     // success!
                     FileManager.PayloadSendSuccess(payload);
                 }
-                else if (req.responseCode >= 500)
+                else if (IsRetryableNow(req.responseCode))
                 {
                     // Something is wrong with the server/connection, retry after a delay
                     DelayBeforeDelivery = true;
                     Send(payload);
                 }
-                else
+                else if(ShouldCache(req.responseCode))
                 {
                     // sending failed, cache payload to disk
                     FileManager.SendPayloadFailed(payload);
                 }
             }
         }
 
+        private bool IsSuccess(long code)
+        {
+            return code == 200 || code == 202;
+        }
+
+        private bool IsRetryableNow(long code)
+        {
+            return code >= 500 || code == 408 || code == 429;
+        }
+
+        private bool ShouldCache(long code)
+        {
+            return code < 400;
```

Comment:
When would this ever happen?  Should this be `< 500`?

Reply:
It makes sense in the context of the code handling the request. This is the last check after we have alread processed 200s 400s and 500s. So basically, if after all that, there is any error that's not a 400, like a 0 for example, then we should cache and try on next launch because it's most likely some kind of network issue.

Reply:
We haven't processed "400s" though we've specifically processed 408 and 429.

Reply:
Ah ok.  There's not really any such thing as ResponseCode 0; I guess it's just a convention.  There's a `response` property on the `UnityWebRequest` I wonder if we could use that instead, just to be clear.

Even if we don't do that, I'd suggest we specifically check for `0` here as it'd be much clearer what the intent is.